var/spack/repos/builtin/packages/fftx/package.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import subprocess
+
+from spack import *
+
+
+class Fftx(CMakePackage):
+    """FIXME: Put a proper description of your package here."""
+
+    homepage = "https://spiral.net"
+    url      = "https://github.com/spiral-software/fftx/archive/0.9.0.tar.gz"
+    git      = "https://github.com/spiral-software/fftx.git"
+
+    maintainers = ['spiralgen']
+
+    version('develop', branch='develop')
+    version('main',  branch='main')
+    version('0.9.0', sha256='d4930e9b959fd56bb63543b359ca09a7ae2c56db3ffc28d7d3628d92fc79ce12')
+
+    variant('build_type', default='Release',
+            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'),
+            description='The build type to build')
+
+    depends_on('spiral-software')
+    #  depends_on('spiral-package-fftx')
+    #  depends_on('spiral-package-simt')
+    #  depends_on('spiral-package-mpi')
+
+    @run_before('cmake')
+    def create_lib_source_code(self):
+        #  From directory examples/library run the build-lib-code.sh script
+        with working_dir(join_path(self.stage.source_path, 'examples/library')):
+            bld_script = join_path(self.stage.source_path,
+                                   'examples/library/build-lib-code.sh')
+            result = subprocess.run(bld_script, shell=True, check=False)
+            res = result.returncode
+            print(result)
+            #  Use the script result to set the build parameter for use with cmake
+            if res == 0:
+                self.build_config = '-D_codegen=CPU'
+            elif res == 1:
+                self.build_config = '-D_codegen=CUDA'
+            elif res == 2:
+                self.build_config = '-D_codegen=HIP'
+
+    def cmake_args(self):
+        spec = self.spec
+        args = [
+            '-DSPIRAL_HOME:STRING={0}'.format(spec['spiral-software'].prefix)
+        ]
+        args.append('-DCMAKE_INSTALL_PREFIX:PATH={0}'.format(self.stage.source_path))
+        args.append(self.build_config)
+        print('Args = ' + str(args))
+        return args
+
+    def build(self, spec, prefix):
+        with working_dir(self.build_directory):
+            make('install')
```

Comment:
@broderickpt wrote:

> Building FFTX depends on SPIRAL and it’s helper packages.  A SPIRAL installation (at least outside of spack and modules) is found based on the SPIRAL_HOME environment variable.  It may be that the default build() would work once the args are defined?

Reply:
Yep I think it would.

See the docs on `CMakePackage` and how to use it here: https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html

Take a look at `range-v3` for a reasonable `CMakePackage` example -- it uses `build_targets` and has a fairly well fleshed out `cmake_args` method.

Reply:
Now setting build_targets and using the default build()